src/main/java/org/opensearch/security/support/SecurityUtils.java

```diff
@@ -140,4 +149,35 @@ private static String resolveEnvVar(String envVarName, String mode, boolean bc,
             return bc ? Hasher.hash(envVarValue.toCharArray(), settings) : envVarValue;
         }
     }
+
+    public static boolean matchesHostPatterns(WildcardMatcher hostMatcher, InetAddress address, String hostResolverMode) {
+        if (hostMatcher == null || address == null) {
+            return false;
+        }
+        List<String> valuesToCheck = new ArrayList<>(List.of(address.getHostAddress()));
+        if (hostResolverMode != null
+            && (hostResolverMode.equalsIgnoreCase("ip-hostname") || hostResolverMode.equalsIgnoreCase("ip-hostname-lookup"))) {
+            final String hostName = address.getHostName();
+            valuesToCheck.add(hostName);
+        }
+
+        return valuesToCheck.stream().anyMatch(hostMatcher);
+    }
+
+    public static boolean matchesCidrPatterns(ClientBlockRegistry<InetAddress> clientBlockRegistry, InetAddress address) {
+        if (clientBlockRegistry == null || address == null) {
+            return false;
+        }
+        AuthFailureListener authFailureListener = (AuthFailureListener) clientBlockRegistry;
+        String hostAddress = address.getHostAddress();
+        return authFailureListener.getIgnoreHosts().parallelStream().filter(pattern -> pattern.indexOf('/') != -1).anyMatch(pattern -> {
```

Comment:
Is using a parallelStream really justified here? AFAIK, it should be rather used in situations where the operation per-item operation clearly outweighs the cost of the thread management. I have doubts whether CIDR matching is really so heavy. 

See: https://www.baeldung.com/java-when-to-use-parallel-stream#1-the-overhead

Reply:
Thanks, moved to ``streams``